### PR TITLE
Typecheck visitor and basic SymbolTable

### PR DESCRIPTION
**Pull request by ginabolo**

### CODE REVIEW

File: src/ast/Action.ts [hunk 1]
```diff
@@ -1,7 +1,17 @@
 import Value from "./Value";
 import ASTVisitor from "../visitor/ASTVisitor";
+import Class from "./class/Class";
+import Parameter from "./Parameter";
 
 export default class Action extends Value {
+    public readonly clss: Class;
```
Comment (JeffreyDoyle) [inline src/ast/Action.ts]: class

File: src/symbol_table/SymbolTable.ts [hunk 2]
```diff
@@ -1,46 +1,106 @@
...
+    }
+
+    public accessActionDefinitionFromFlow(varName: string, flowName:string): Action{
+        // check inside flow first otherwise global
+        let f = this.accessFlow(flowName);
+
+        try {
+            return f.accessAction(varName);
+        } catch (e){
+            this.accessAction(varName);
```
Comment (kizggerg) [inline src/symbol_table/SymbolTable.ts]: Need return statement

File: src/visitor/TypeCheckVisitor.ts [hunk 3]
```diff
@@ -1,8 +1,225 @@
...
+        }
+    }
+
+    visitFlow(flow: AST.Flow): any {
+        // recurse on declarations and modifiers. Check that declarations do not include flows
+        if(flow.declarations.filter((dec) => dec.type instanceof AST.FlowType).length > 0){
+            throw new Error("Illegal nested flow.");
+        }
+
+        for (let dec in flow.declarations){
```
Comment (kizggerg) [inline src/visitor/TypeCheckVisitor.ts]: flow.declarations.map((dec) => dec.acceptASTVisitor(this));

**Comment by ginabolo** [comment]
@kizggerg I believe this addresses all your feedback

**Comment by kizggerg** [comment]
LGTM!

**Pull request merged**